internal/connect/collectors.go

```diff
@@ -4,24 +4,38 @@ import (
 	"github.com/SUSE/connect-ng/internal/collectors"
 )
 
-// Collectors which are to be used when fetching the system information.
-var usedCollectors = []collectors.Collector{
-	collectors.CPU{},
-	collectors.Hostname{},
-	collectors.Memory{},
-	collectors.UUID{},
-	collectors.Virtualization{},
-	collectors.CloudProvider{},
-	collectors.Architecture{},
-	collectors.ContainerRuntime{},
-
-	// Optional collectors
-	collectors.Uname{},
-	collectors.SAP{},
+// Fetch system information based on the available collectors for the system.
+func FetchSystemInformation(infoOnly bool) (collectors.Result, error) {
+	var usedCollectors = []collectors.Collector{
+		collectors.CPU{},
+		collectors.Hostname{},
+		collectors.Memory{},
+		collectors.UUID{},
+		collectors.Virtualization{},
+		collectors.CloudProvider{},
+		collectors.Architecture{},
+		collectors.ContainerRuntime{},
+
+		// Optional collectors
+		collectors.Uname{},
+		collectors.SAP{},
+	}
+
+	arch, err := collectors.DetectArchitecture()
+
+	if err != nil {
+		return collectors.NoResult, err
+	}
+	return collectors.CollectInformation(arch, usedCollectors)
 }
 
-// Fetch system information based on the available collectors for the system.
-func FetchSystemInformation() (collectors.Result, error) {
+// Fetch system profile information
+func FetchSystemProfiles(infoOnly bool) (collectors.Result, error) {
+	var usedCollectors = []collectors.Collector{
+		collectors.PCI{UpdateDataIDs: !infoOnly},
+		collectors.LSMOD{UpdateDataIDs: !infoOnly},
+	}
+
 	arch, err := collectors.DetectArchitecture()
```

Comment:
Note that this was already called by FetchSystemInformation. Hence, why not add a "arch" flag to this function's signature?